lightning/src/ln/onion_utils.rs

```diff
@@ -901,25 +915,40 @@ fn build_unencrypted_failure_packet(
 		res.extend_from_slice(&failure_data[..]);
 		res
 	};
+	let pad_len = if failure_data.len() > 1024 - 2 { 0 } else { 1024 - 2 - failure_data.len() };
```

Comment:
Reminder: need to parameterize this for the test. Maybe no need to bump it from 256 otherwise.